aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -87,7 +87,8 @@ void batch_norm_elementwise(
   case Impl::ChannelsLast: {
     auto weight = at::borrow_from_optional_tensor(weight_opt);
     auto bias = at::borrow_from_optional_tensor(bias_opt);
-    if ((!weight->defined() || weight->is_contiguous()) &&
+    if (self->is_contiguous() &&
```

Comment:
I don't think that's right. `self.is_contiguous(at::MemoryFormat::ChannelsLast)` is already checked in `batch_norm_choose_impl`, but `output` is just assumed to be channels-last contiguous in the kernel, and we need to check for it here. Does it make sense? I think Impl::Contiguous also is not properly checking contiguity of `out`.

Reply:
Right, this makes sense as I was getting the ordering confused between `batch_norm_cuda_out` and `batch_norm_elementwise`. For a moment I thought the kernel call was in `batch_norm_cuda_out`.